src/main/kotlin/objects/Game.kt

```diff
@@ -21,60 +21,61 @@ class Game(val p1Connection: GameConnection, val p2connection: GameConnection, d
         if (player == Player.Player1)
             connection.sendPacket(StartTurnPacket())
         while (connection.isOpen) {
-            when (val packet = connection.receivePacket()) {
-                null -> {
-                    if (connection.isOpen)
-                        connection.close()
-                    println(prefix + "Connection was closed unexpectedly")
-
-                    if (otherConnection.isOpen) {
-                        println(prefix + "Informing opponent")
-                        otherConnection.sendPacket(
-                            DisconnectPacket(
-                                DisconnectPacket.Reason.opponent_disconnect,
-                                "The opponent has closed it's connection"
+            boardManager.withGameOverHandler {
+                when (val packet = connection.receivePacket()) {
+                    null -> {
```

Comment:
In this case the connection is closed so running the code in `withGameOverHandler` will cause errors.